.github/workflows/codeql.yml

```
@@ -0,0 +1,64 @@
+name: "CodeQL Advanced"
+
+on:
+  push:
+    branches: [ "develop", "release/*" ]
+  pull_request:
+    branches: [ "develop", "release/*" ]
+  schedule:
+    - cron: '38 2 * * 2'
+  workflow_call:
+    inputs:
+      ruby_version:
+        description: 'Ruby Version'
+        default: "3.3.4"
+        type: string
+        required: false
+      node_version:
+        description: 'Node version'
+        default: '22.14.0'
+        required: false
+        type: string
+
+jobs:
+  analyze:
+    name: Analyze (${{ matrix.language }})
+    runs-on: ubuntu-22.04
```

We should run just of the last commit. 
```suggestion
    runs-on: ubuntu-22.04
    timeout-minutes: 60
    concurrency:
      group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}-${{ matrix.language }}
      cancel-in-progress: true
```